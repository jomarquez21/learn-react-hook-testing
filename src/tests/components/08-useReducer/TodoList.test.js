import React from 'react';
import {shallow} from 'enzyme';
import {TodoList} from './../../../components/08-useReducer/TodoList';
import {demoTodos} from './../../fixtures/demoTodos'

describe('Test on `TodoList`', () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const component = shallow(<TodoList todos={demoTodos} handleDelete={handleDelete} handleToggle={handleToggle} />);
  
  test('Debe mostrarse correctamente', () => {
    expect(component).toMatchSnapshot()
  });

  test('Debe tener dos <TodoListItem />', () => {
    expect(component.find('TodoListItem').length).toBe(demoTodos.length);

    expect(component.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
  });
})