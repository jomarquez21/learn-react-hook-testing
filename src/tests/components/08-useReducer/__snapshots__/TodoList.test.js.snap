// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on \`TodoList\` Debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Learn english",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Learn react",
        "done": true,
        "id": 2,
      }
    }
  />
</ul>
`;
